plugins/obs-filters/data/drop_shadow_filter.effect

```diff
@@ -0,0 +1,190 @@
+uniform float4x4 ViewProj;
+uniform texture2d image;
+
+uniform float2 image_uv_offset;
+uniform float2 image_uv_scale;
+uniform float2 image_uv_pixel_interval;
+uniform float2 shadow_uv_pixel_interval;
+
+uniform float2 offset;
+uniform float4 shadow_color;
+uniform texture2d shadow;
+
+uniform float kawase_distance;
+
+sampler_state textureSampler {
+	Filter    = Linear;
+	AddressU  = Border;
+	AddressV  = Border;
+	BorderColor = 00000000;
+};
+
+struct VertData {
+	float4 pos : POSITION;
+	float2 shadow_uv : TEXCOORD0;
+};
+
+VertData VSDefault(VertData v_in)
+{
+	VertData vert_out;
+	vert_out.pos = mul(float4(v_in.pos.xyz, 1.0), ViewProj);
+	vert_out.shadow_uv = v_in.shadow_uv;
+	return vert_out;
+}
+
+float4 SampleImageRGBA(float2 image_uv)
+{
+	return image.Sample(textureSampler, image_uv);
+}
+
+float SampleImageAlpha(float2 image_uv)
+{
+	return image.Sample(textureSampler, image_uv).a;
+}
+
+float SampleShadow(float2 image_uv)
+{
+	return shadow.Sample(textureSampler, image_uv).a;
+}
+
+float SampleShadowFromImage(float2 image_uv)
+{
+	float2 uv = image_uv + image_uv_pixel_interval * offset;
+	float2 distance = image_uv_pixel_interval *
+			float2(kawase_distance, kawase_distance);
+
+	float x = 0.0;
+	x += SampleImageAlpha(uv + distance * float2(-1.0, -1.0));
+	x += SampleImageAlpha(uv + distance * float2( 1.0, -1.0));
+	x += SampleImageAlpha(uv + distance * float2( 1.0,  1.0));
+	x += SampleImageAlpha(uv + distance * float2(-1.0,  1.0));
+
+	return x * 0.25;
+}
+
+float SampleShadowFromShadow(float2 shadow_uv)
+{
+	float2 distance = shadow_uv_pixel_interval *
+			float2(kawase_distance, kawase_distance);
+
+	float x = 0.0;
+	x += SampleShadow(shadow_uv + distance * float2(-1.0, -1.0));
+	x += SampleShadow(shadow_uv + distance * float2( 1.0, -1.0));
+	x += SampleShadow(shadow_uv + distance * float2( 1.0,  1.0));
+	x += SampleShadow(shadow_uv + distance * float2(-1.0,  1.0));
+
+	return x * 0.25;
+}
+
+// This function is correct for both Rec.601 and Rec.709
+float4 LinearToRec(float4 x)
+{
+	float3 inv_gamma = float3(0.45, 0.45, 0.45);
+
+	// Remove alpha pre-multiplication.
+	float3 color = x.rgb / x.a;
+	float3 low = 4.500 * color;
+	float3 high = 1.099 * pow(color, inv_gamma) - 0.099;
+
+	return float4(
+		x.r < 0.018 ? low.r : high.r,
+		x.g < 0.018 ? low.g : high.g,
+		x.b < 0.018 ? low.b : high.b,
+		x.a
+	);
+}
+
+// This function is correct for both Rec.601 and Rec.709
+float4 RecToLinear(float4 x)
+{
+	float3 gamma = 1.0 / float3(0.45, 0.45, 0.45);
+
+	float3 low = x.rgb / 4.500;
+	float3 high = pow((x.rgb + 0.099) / 1.099, gamma);
+	float3 color = float3(
+		x.r < 0.081 ? low.r : high.r,
+		x.g < 0.081 ? low.g : high.g,
+		x.b < 0.081 ? low.b : high.b
+	);
+
+	// Pre multiply alpha.
```

Comment:
Spelling:  Use either "premultiply" or "pre-multiply".  Other variants of "premultiply" occur throughout here, so you should probably keep them all consistent.